### PR TITLE
Refactor FLAC Converter

### PR DESCRIPTION
**Pull request by cbj4074**
The intention herein is to refactor the remaining legacy methods in `FlacConverter` (namely, `toAlac()`).

This requires adding a second audio format to the suite, which in turn requires adding a second CLI tool (`ffmpeg`), in addition to the one that we first added (`sox`). Doing so introduces a with it a handful of challenges to be discussed and addressed.

### CODE REVIEW

File: src/Flac/FlacConverter.php [hunk 1]
```diff
@@ -51,119 +56,21 @@ public function setSupportedOutputFormats(array $supportedOutputFormats): void
...
-        // If "['LC_ALL' => 'en_US.utf8']" is not passed here, any UTF-8 character will appear as a "#" symbol.
+        // If "['LC_ALL' => 'en_US.utf8']" is not passed here, any UTF-8
+        // character will appear as a "#" symbol.
 
         $env = ['LC_ALL' => 'en_US.utf8'];
 
-        $this->process->setCommand($this->effect->getCommand()->compose());
+        $this->process->setCommand($this->flacEffect->getCommand()->compose());
 
         $this->process->setTimeout(600);
```
Comment (poppabear8883) [inline src/Flac/FlacConverter.php:75]: Why this ?

Comment (cbj4074) [reply src/Flac/FlacConverter.php:75]: This is precisely what I wanted to discuss.

If you look at the `toAlac()` method body, you will see `$this->alacEffect->...`.

Basically, we have the need to use two different `Effect` classes within the `FlacConverter` class: one for FLAC files and one for ALAC files.

Prior to this commit, we had only needed one (for FLAC files), so `$this->effect->...` was sufficient.

I'm struggling to visualize a better solution to this need...

We discussed this while brainstorming in a side-channel and made some notes in the Effect class constructors, e.g.: https://github.com/indiehd/audio-manipulator/blob/5465c98398612f34c3c2f6b2e155aba12da3a00b/src/Alac/AlacEffects.php#L17

I feel like I'm missing something obvious here. Throw me a bone? :D

Comment (cbj4074) [reply src/Flac/FlacConverter.php:75]: Resolving (pardon the pun!) this issue seems better-suited for a separate PR. If you're agreeable to it, let's resolve this conversation and address this in a subsequent PR.

**Review by poppabear8883 — APPROVED** [review]
LGTM

**Review by poppabear8883 — APPROVED** [review]

**Pull request merged**